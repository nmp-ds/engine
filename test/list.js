import { setup } from "./_helpers.js";
import { describe, expect, test } from "vitest";

setup();

describe("list", () => {
  test('list style type', async ({ uno }) => {
    const classes = [
      'list-disc',
      'list-circle',
      'list-square',
      'list-decimal',
      'list-zero-decimal',
      'list-none',
    ];
    const { css } = await uno.generate(classes);
    expect(css).toMatchSnapshot();
  });

  test('list style position', async ({ uno }) => {
    const { css } = await uno.generate(['list-outside', 'list-inside']);
    expect(css).toMatchSnapshot();
  });

  test('list style invalid class', async ({ uno }) => {
    const { css } = await uno.generate(['list-style-invalid', 'list', 'list-[decimal]']);
    expect(css).toMatchInlineSnapshot('""');
  });

  test("Render style for list-checked", async (t) => {
    const classes = ['list-checked'];

    const { css } = await t.uno.generate(classes);

    expect(css).toMatchInlineSnapshot(`
      "/* layer: default */
      .list-checked{line-height:var(--w-font-line-height-1);}.list-checked>li{position:relative;padding-left:24px;}.list-checked>li::before{content:\\"\\";display:block;position:absolute;width:16px;height:28px;left:0;color:var(--w-list-color-icon-checked);background-size:contain;background-position:50%;background-repeat:no-repeat;background-image:\\"\\";}"
    `);
  });

});
