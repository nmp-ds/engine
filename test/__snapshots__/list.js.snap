// Vitest Snapshot v1

exports[`list > list style position 1`] = `
"/* layer: default */
.list-outside{list-style-position:outside;}
.list-inside{list-style-position:inside;}"
`;

exports[`list > list style type 1`] = `
"/* layer: default */
.list-circle{list-style-type:circle;}
.list-decimal{list-style-type:decimal;}
.list-disc{list-style-type:disc;}
.list-square{list-style-type:square;}
.list-zero-decimal{list-style-type:decimal-leading-zero;}
.list-none{list-style-type:none;}"
`;
